addons/editor-number-arrow-keys/userscript.js

```diff
@@ -0,0 +1,31 @@
+export default async function ({ addon }) {
+  setInterval(() => {
+    [...document.querySelectorAll(".blocklyHtmlInput:not(.sa-editor-number-arrow-keys)")].forEach((elt) => {
+      elt.addEventListener("keydown", (e) => {
```

Comment:
No need to use `setInterval`, which won't be nice for performance, and no other addons use it this way.

Simply register (once!) a single `keydown` event on `document.body` and check if `e.target.classList.has("blocklyHtmlInput")`.
This also removes the need for a `sa-editor-number-arrow-keys` class.